src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -501,39 +498,39 @@ internal void ExecutePackageDownload(string name, PackageVersion version, string
             var pmExt = DynamoViewModel.Model.GetPackageManagerExtension();
             if (result == MessageBoxResult.OK)
             {
-                // get all of the headers
-                var headers = version.full_dependency_ids.Select(dep => dep._id).Select((id) =>
+                // get all of the dependency version headers
+                var dependencyVersionHeaders = version.full_dependency_ids.Select((dep, i) =>
                 {
-                    PackageHeader pkgHeader;
-                    var res = Model.DownloadPackageHeader(id, out pkgHeader);
-
-                    if (!res.Success)
-                        MessageBox.Show(String.Format(Resources.MessageFailedToDownloadPackage, id),
+                    try
+                    {
+                        var depVersion = version.full_dependency_versions[i];
+                        var res = Model.GetPackageVersionHeader(dep._id, depVersion);
+                        return res;
+                    }
+                    catch
+                    {
+                        MessageBox.Show(
+                            String.Format(Resources.MessageFailedToDownloadPackageVersion, dep._id),
                             Resources.PackageDownloadErrorMessageBoxTitle,
                             MessageBoxButton.OK, MessageBoxImage.Error);
-
-                    return pkgHeader;
+                        return null;
+                    }
                 }).ToList();
 
                 // if any header download fails, abort
-                if (headers.Any(x => x == null))
+                if (dependencyVersionHeaders.Any(x => x == null))
                 {
                     return;
                 }
 
-                var allPackageVersions = PackageManagerSearchElement.ListRequiredPackageVersions(headers, version);
-
                 // determine if any of the packages contain binaries or python scripts.  
-                var containsBinaries =
-                    allPackageVersions.Any(
-                        x => x.Item2.contents.Contains(PackageManagerClient.PackageContainsBinariesConstant) || x.Item2.contains_binaries);
-
-                var containsPythonScripts =
-                    allPackageVersions.Any(
-                        x => x.Item2.contents.Contains(PackageManagerClient.PackageContainsPythonScriptsConstant));
+                var containsBinariesOrPythonScripts = dependencyVersionHeaders.Any(x => 
+                    x.contents.Contains(PackageManagerClient.PackageContainsBinariesConstant) ||
+                    x.contains_binaries ||
```

Comment:
We would get some minor performance gains by moving the `contains_binaries` test before the `Contains()` check.

Reply:
Agree